app/src/main/java/com/tari/android/wallet/ui/activity/tx/TxDetailsActivity.kt

```diff
@@ -135,18 +143,30 @@ internal class TxDetailsActivity : AppCompatActivity(), ServiceConnection {
         ui = ActivityTxDetailsBinding.inflate(layoutInflater).apply { setContentView(root) }
         overridePendingTransition(R.anim.enter_from_right, R.anim.exit_to_left)
         setupUI()
-        tracker.screen(path = "/home/tx_details", title = "Transaction Details")
-    }
 
-    override fun onStart() {
-        super.onStart()
-        // start service if not started yet
+        val savedTx = savedInstanceState?.getParcelable<Tx>(TX_EXTRA_KEY)
+        val intentTx = intent.getParcelableExtra<Tx>(TX_EXTRA_KEY)
+        when {
```

Comment:
could it be `if (savedTx != null || intentTx != null)` with subsequent `tx = savedTx ?: intentTx`?

Reply:
how about
```kotlin
(savedInstanceState?.getParcelable(TX_EXTRA_KEY)
    ?: intent.getParcelableExtra<Tx>(TX_EXTRA_KEY))?.let {
    tx = it
    bindTxData()
    observeTxUpdates()
    enableCTAs()
}
```

Reply:
or
```kotlin
val savedTx = savedInstanceState?.getParcelable<Tx>(TX_EXTRA_KEY)
val intentTx =  intent.getParcelableExtra<Tx>(TX_EXTRA_KEY)
(savedTx ?: intentTx)?.let {
    tx = it
    bindTxData()
    observeTxUpdates()
    enableCTAs()
}
```